sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1812,7 +1812,13 @@ class Dataset[T] private[sql](
    * @since 2.0.0
    */
   def dropDuplicates(colNames: Seq[String]): Dataset[T] = withTypedPlan {
-    val groupCols = colNames.map(resolve)
+    val resolver = sparkSession.sessionState.analyzer.resolver
+    val groupCols = colNames.map {
```

Comment:
this can be:
```
val allColumns = queryExecution.analyzed.output
val groupCols = colNames.map { c =>
  allColumns.find(col => resolver(col.name, c)).getOrElse(throw new AnalysisException(...))
}
```